context/app/static/js/components/workspaces/WorkspacesList.jsx

```diff
@@ -1,25 +1,39 @@
-import React from 'react';
+import React, { useRef } from 'react';
 import Paper from '@mui/material/Paper';
 import Typography from '@mui/material/Typography';
+import Stack from '@mui/material/Stack';
+import DeleteRounded from '@mui/icons-material/DeleteRounded';
+import AddRounded from '@mui/icons-material/AddRounded';
 
 import Description from 'js/shared-styles/sections/Description';
-import { DeleteIcon } from 'js/shared-styles/icons';
 import { SpacedSectionButtonRow } from 'js/shared-styles/sections/SectionButtonRow';
-import { PanelWrapper } from 'js/shared-styles/panels';
 
-import WorkspaceDetails from 'js/components/workspaces/WorkspaceDetails';
-import CreateWorkspaceButton from 'js/components/workspaces/CreateWorkspaceButton';
+import WorkspaceListItem from 'js/components/workspaces/WorkspaceListItem';
+import { useSelectItems } from 'js/hooks/useSelectItems';
+import { useSnackbarActions } from 'js/shared-styles/snackbars';
 import { useWorkspacesList } from './hooks';
-import { StyledButton } from './style';
+import WorkspaceButton from './WorkspaceButton';
 
 function WorkspacesList() {
-  const {
-    workspacesList,
-    handleDeleteWorkspace: handleDelete,
-    handleCreateWorkspace,
-    handleStopWorkspace: handleStop,
-    handleStartWorkspace,
-  } = useWorkspacesList();
+  const { workspacesList, handleCreateWorkspace, handleDeleteWorkspace } = useWorkspacesList();
+
+  const { selectedItems, toggleItem } = useSelectItems();
+  const { toastError } = useSnackbarActions();
+
+  const isDeleting = useRef(false);
+
+  const handleDeleteSelected = async () => {
+    const workspaceIds = Array.from(selectedItems);
```

Comment:
Benefit of `Array.from` versus spreading?

Reply:
After looking into it, it looks like spreading is actually a bit more performant than Array.from on arrays of under 10 elements, and equal in performance above that, so I've swapped this over to a spread.

![](https://miro.medium.com/v2/resize:fit:640/format:webp/1*TaYfksKo9LUsu3oKGBlzVw.png)